src/passes/OptimizeInstructions.cpp

```diff
@@ -1345,15 +1381,57 @@ struct OptimizeInstructions
         }
       }
       // operations on all 1s
-      // TODO: shortcut method to create an all-ones?
       if (right->value == Literal(int32_t(-1)) ||
           right->value == Literal(int64_t(-1))) {
         if (binary->op == Abstract::getBinary(type, Abstract::And)) {
+          // x & -1   ==>   x
           return binary->left;
         } else if (binary->op == Abstract::getBinary(type, Abstract::Or) &&
                    !EffectAnalyzer(getPassOptions(), features, binary->left)
                       .hasSideEffects()) {
+          // x | -1   ==>   -1
           return binary->right;
+        } else if (binary->op == Abstract::getBinary(type, Abstract::Sub)) {
+          // x - (-1)   ==>   x + 1
```

Comment:
don't we have a place that handles `x - (-y) => x + y` for all y and not just a constant 1?

(this looks wrong here, it's much too specific to `1`)

Reply:
Yeah, definitely need generalize it to any rhs

Reply:
Will do later I currently fuzzing this recent PR #2884

Reply:
I removed this transform at all due to potential conflict with [this](https://github.com/WebAssembly/binaryen/pull/2870/files#diff-54e02686df3480c98e3f90a24705c00bR1431) normalization